catalog/core/catalog-core-api/src/main/java/ddf/catalog/validation/ReportingMetacardValidator.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.validation.report.MetacardValidationReport;
+
+/**
+ * Validates a {@link Metacard} and provides information via a report if problems exist with the
+ * {@link Metacard}'s data.
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface ReportingMetacardValidator {
+    /**
+     * Validates a {@link Metacard}.
+     *
+     * @param metacard the {@link Metacard} to validate
+     * @return a {@link MetacardValidationReport} describing the outcome
```

Comment:
I'm wondering, how would a client know that there was a validation problem? Do they need to check that the `ValidationViolation` list is empty? If so, should we instead provide some kind of boolean method to check, e.g., `MetacardValidationReport.hasViolations()`, or use Java 8's `Optional`? I personally think the latter looks better :-)